js/packages/web/src/actions/sendPlaceBid.ts

```diff
@@ -47,17 +47,33 @@ export async function sendPlaceBid(
     signers,
   );
 
-  const res = await sendTransactionWithRetry(
-    connection,
-    wallet,
-    instructions[0],
-    signers[0],
-    commitment,
-  );
-
-  if (res.err) throw res.err.inner;
-
-  const { txid } = res;
+  let bidError: Error | null = null;
+  let txid = '';
+
+  await SmartInstructionSender.build(wallet as any, connection)
+    .config({
+      abortOnFailure: true,
+      maxSigningAttempts: 3,
+      commitment: commitment,
+    })
+    .withInstructionSets(
+      instructions.map((ixs, i) => ({
+        instructions: ixs,
+        signers: signers[i],
+      })),
+    )
+    .onFailure(err => {
+      bidError = err;
+    })
+    .onProgress((_, x) => {
+      txid = x;
+      console.log('transaction ID:', x);
+    })
+    .send();
+
+  if (bidError) {
+    throw bidError;
+  }
```

Comment:
Something about this block tells me that you should rather do an `await new Promise((resolve, reject) => ...)` and resolve/reject in the onProgress if it's the final one and the onFailure blocks, but this probably works as it is right now due to the await on the call. Further testing required ?